### PR TITLE
Fixed may cause wrong recovery position

### PR DESCRIPTION
**Pull request by MrLittleWhite**
When m_position is set, m_bytesRead should be reset to 0. 

                        /*
                         * Try to recover gracefully if we have a non-continuous stream
                         */
                        Input_Stream_Position currentPosition = THIS->position();
                        
                        Input_Stream_Position recoveryPosition;
                        recoveryPosition.start = currentPosition.start + THIS->m_bytesRead;
                        recoveryPosition.end = THIS->contentLength();
                        
                        HS_TRACE("Recovering HTTP stream, start %llu\n", recoveryPosition.start);
                        
                        /*if error occurred, stop timer, reset m_isReadedData */
                        THIS->resetOpenTimer(true);
                        
                        THIS->open(recoveryPosition);

If not ,when it execute the code above in http_stream.cpp, may cause wrong recoveryPosition.

### CODE REVIEW

**Comment by MrLittleWhite** [comment]
New feature is added: FreeStream can play When network lost. HTTP stream will reopen after network is available again.

File: FreeStreamer/FreeStreamer/audio_stream.cpp [hunk 1]
```diff
@@ -1441,6 +1456,14 @@ void Audio_Stream::seekTimerCallback(CFRunLoopTimerRef timer, void *info)
         THIS->m_converterRunOutOfData = false;
         THIS->m_discontinuity = true;
         
+        // check if inputStream has error
+        if (THIS->m_inputStream) {
+            CFStringRef errString = THIS->m_inputStream->errorDescription();
+            THIS->m_inputStream->close();
+            THIS->closeAndSignalError(AS_ERR_NETWORK, errString);
+            return;
+        }
```
Comment (dfg-nota) [inline FreeStreamer/FreeStreamer/audio_stream.cpp]: Shouldn't there be a check on errString before closing?

Comment (MrLittleWhite) [inline FreeStreamer/FreeStreamer/audio_stream.cpp]: Yeah, I do check on line 823,but forget this one. Thanks. It's fixed now.

**Comment by muhku** [comment]
This PR should be split into multiple PRs. I don't think all the commits are related to the fix.

**Pull request closed without merging**